packages/mobile/src/escrow/utils.ts

```diff
@@ -31,3 +35,14 @@ export const generateUniquePaymentId = (
     }
   }
 }
+
+export const useEscrowPaymentRecipientName = (payment: EscrowedPayment) => {
+  const { recipientPhone, recipientIdentifier } = payment
+
+  const identifierToe164Number = useSelector(identifierToe164NumberSelector)
+  const recipientCache = useSelector(recipientCacheSelector)
+  const phoneNumber = identifierToe164Number[recipientIdentifier] ?? recipientPhone
```

Comment:
Isn't this part redundant? IIUC `recipientPhone` is derived exclusively from `identifierToe164Number` in `escrow/saga` so it may not be worthwhile to import it again here

Reply:
The reason I did it this way is that the saga is run once and is not updated if we fetch the pepper after it's called, the only way to see the name in that case is to pull to refresh or restart the app. Doing it this way will show the name as soon as the pepper for the number is fetched.

For most cases this doesn't make a difference, it only matters for users who import their account so they don't have any peppers fetched and they might fetch them after doing the initial load of pending escrows.

Reply:
Oh I see. In that case is `identifierToe164Number[recipientIdentifier]` ever `undefined` while `recipientPhone` is defined?

Reply:
Yes, we could remove the `recipientPhone` from the escrow payment. It requires changing a screen or two more and those changes will conflict with the CIP-8 PR. I added a TODO to remove the phone number from the escrow payment object so that we always fetch it from the mapping, this is something we can do during the invite redesign work coming up this milestone.

Reply:
Cool makes sense